controllers/cloudstackmachine_controller.go

```diff
@@ -223,15 +229,18 @@ func processCustomMetadata(data []byte, r *CloudStackMachineReconciliationRunner
 // ConfirmVMStatus checks the Instance's status for running state and requeues otherwise.
 func (r *CloudStackMachineReconciliationRunner) RequeueIfInstanceNotRunning() (retRes ctrl.Result, reterr error) {
 	if r.ReconciliationSubject.Status.InstanceState == "Running" {
+		r.Recorder.Event(r.ReconciliationSubject, "Normal", "Running", "Machine instance is Running...")
 		r.Log.Info("Machine instance is Running...")
 		r.ReconciliationSubject.Status.Ready = true
 	} else if r.ReconciliationSubject.Status.InstanceState == "Error" {
+		r.Recorder.Event(r.ReconciliationSubject, "Warning", "Error", "CloudStackMachine VM in error state. Deleting associated Machine")
 		r.Log.Info("CloudStackMachine VM in error state. Deleting associated Machine.", "csMachine", r.ReconciliationSubject.GetName())
 		if err := r.K8sClient.Delete(r.RequestCtx, r.CAPIMachine); err != nil {
 			return ctrl.Result{}, err
```

Comment:
should we capture this error as an event as well?

Reply:
Ah never mind, I thought this call to `Delete` was to CloudStack, not the K8sClient. I think we can disregard.